### DISCUSSION TITLE
Does IPSec work with VLAN / VxLAN

### DISCUSSION THREAD

**Discussion by lijh8** [strongswan/strongswan#621] [General]
In Site-Site case, e.g. there're two PCs: PC1 in moon-net and PC2 in sun-net.

Say our client applied VLAN / VxLAN for all their PCs.

1. If PC1 and PC2 are in same VLAN / VxLAN, can they reach each other and get protected by Strongswan?

2. If PC1 and PC2 are *NOT* in same VLAN / VxLAN, can they reach each other and get protected by Strongswan?

Thanks

**Comment by Thermi** [comment]
1. depends on exact network configuration and if you use virtual IPs. Generally, should be possible.
2. Yes.
